src/sentry/incidents/subscription_processor.py

```diff
@@ -550,8 +551,29 @@ def trigger_alert_threshold(
         :return:
         """
         self.trigger_alert_counts[trigger.id] += 1
+
+        # If an incident was created for this rule, trigger type, and subscription
+        # within the last 10 minutes, don't make another one
+        last_incident: Incident | None = (
+            Incident.objects.filter(alert_rule=trigger.alert_rule).order_by("-date_added").first()
```

Comment:
@wedamija do you think we need an index to avoid scanning every incident for a given alert rule?

Reply:
Yeah, we will likely need an index on `alert_rule, date_added`